### PR TITLE
deletes the security cyborg

### PR DESCRIPTION
**Pull request by SomeguyManperson**
### There is now a player poll on the server to gather opinion on this PR, the vote is non binding meaning the PR may be merged or closed regardless of the outcome of the vote.

Imagine having a borg module that is so combat capable that all other borg modules are rendered useless
:cl:  
rscdel: deletes the security cyborg
/:cl:

### CODE REVIEW

**Comment by LoliconSlayer** [comment]
Heeere we go

**Comment by Sir-Lagsalot** [comment]
👌

**Comment by Hopekz** [comment]
**Disclaimer: not a meme PR.**

**Comment by SomeguyManperson** [comment]
yes I'm verified what about it

**Comment by monster860** [comment]
"verified" is a fancy term for "I'm using the web editor"

**Comment by DLCabbose** [comment]
no fuck off and die

**Comment by maxion12345** [comment]
This is gonna go just like xenoling, where we delete a big part of something without tweaking it or something first and it practically kills the thing. At least have something to replace it, or fuckin attempt to look at tweaks to it first. And what exactly is wrong with having combat oriented borgs? Perhaps it's because AIs frequently spawn with lawsets enabling them to valid, that this is an issue. Combat modules are also the bread and butter of antag AIs in many cases. And, perhaps, we can look at what makes the secborg so proficient at combat, and simply tweak that. You know how engiborgs have magnetic treads or whatever? Secborgs could just be slow. Makes it hard to valid the traitors if they can just outrun you. And is it the stunbaton that's the cause of the secborgs proficient combat skills? Why don't we look at that, instead of looking at something and deleting it instantly. I feel like we're treating a symptom and not the diease here.

**Comment by Ratvar** [comment]
1. Xenoling wasn't a big part of xenobio
2. Xenoling was tweaked in attempt to balance it, actually, which failed because entire concept was bad for the game, no matter if and how nerfed. I guess 1. Adding a big red warning to follow your laws instead of validhunting 2. Big Taser Removal rework 3. Cooldown to disabler were attempts to balance secborgs as well, if split across the timeline
3. You're free to propose something to replace secborgs, even code it. Someone wanted detective borgs, I've heard. That too was an issue with xenolings: xenomains who liked it were entirely satisfied with how broken it was, or they wouldn't like it in the first place, and so proposed no alternative that would fix the issue of "Afk 30 minutes in xenobio, then validhunt with antag powers and attack officers who try to stop you for breaking laws and rules"
4. Killing a thing isn't bad if said thing is bad 

Really, how would you tweak it? What would you replace secborgs with?
I would remove EMP stun, make it confuse borgs instead, or disable their modules: EMP becomes a disengaging tool that guarantees you can escape, but doesn't instakill all silicons. This way silicons don't need security module as much
And you're still free to propose a **xenoling** replacement: never late

**Comment by Alexandria404** [comment]
nice meme PR

**Comment by Hopekz** [comment]
> nice meme PR

This isn't a meme PR. 
State your argument if you are pro keeping it.
Every other major codebase has already removed their sec borgs.

**Review by ThatLing — CHANGES_REQUESTED** [review]
There's a config option to disable sec borg, change that instead of just removing the option :v

**Comment by Mr-Space** [comment]
nice fucking meme

**Comment by Hopekz** [comment]
> nice fucking meme

To be fair I thought remove xenoling was a meme so I get your denial.
Hopefully we keep getting "nice meme" opposition because at this rate if that's the only pro argument vs the remove argument then secborgs may actually need to get thanos snapped like xenoling.

We're the only major codebase at the moment besides goon with secborgs but goon is wildly different than us imo so I can't even use that as a "pro" point.

**Comment by alexkar598** [comment]
### Several comments on this PR have been hidden due to being offtopic as I suspect this PR will become very controversial very quickly. Binary input(I like/I no like) goes in the reactions by clicking on the thumbs up/thumbs down button respectively. More fletched out arguments can go into comments.

**Comment by alexkar598** [comment]
A server poll will be made to get the general mood of the playerbase towards sec borgs but for the moment it is going to be non binding meaning this PR can be closed or merged despite what the vote says.

**Comment by yacabo** [comment]
This will make it extremely hard for an rogue AI to preform stealth kills

**Comment by nesisknesis** [comment]
> This will make it extremely hard for an rogue AI to preform stealth kills

I wouldn't think so, a clever borg and AI can kill people without the sec module. An electric stun arm that emagged engie borgs have can serve the same function as a stunbaton, and the standard module also has zipties.

Extremely hard is an overstatement, if sec modules are removed, being a rogue AI just requires more thought when it comes to stealth kills.

**Comment by alexkar598** [comment]
### There is now a player poll on the server to gather opinion on this PR, the vote is non binding meaning the PR may be merged or closed regardless of the outcome of the vote.

**Comment by Readystorm546** [comment]
This needs a huge i ded tag but some dumb ass maintainer removed that probably because they died. 
Also i think this is a really bad idea 

> > This will make it extremely hard for an rogue AI to preform stealth kills
> 
> I wouldn't think so, a clever borg and AI can kill people without the sec module. An electric stun arm that emagged engie borgs have can serve the same function as a stunbaton, and the standard module also has zipties.
> 
> Extremely hard is an overstatement, if sec modules are removed, being a rogue AI just requires more thought when it comes to stealth kills.

Most player are not fucking masters at the game and just because you or anyone you know is a master does not mean you should remove something that might help the newer players.

**Comment by maxion12345** [comment]
My ideas to tweak secborg are pretty simple. Let's look at why borgs are so powerful: they got their ranged disabler, and melee stunbaton. They also feel no pain, and fuckin zoom fast. Now, the only thing that makes them different from any normal security guard here is their speed and the fact they can go into negative health. Potential tweaks could include making the Borg slower then other non-combat borgs, or implement Borg slowdown when damaged type of system. And, of course, there's the fact that AI is seen as a way to validhunt by the players. Secborgs are, again, also the bread and butter of a malf ai. You could siphon, but pressure damage takes forever to kill and it feels like everyone carries a welder to weld down vents with now. Shock a door? Only works if they aren't carrying insuls, or a diagnostic hud, or walk into a door. And again, for all the powers a Borg has, its counters are deadly. For instance, the Borg cannot repair burn damage to other borgs, so lasers and fire will put borgs out or action for awhile. Flashes can be found and made in numerous locations, and instantly stun and mute a Borg for a considerable amount of time, and they include an aoe blind attack for you to get close enough for the stun. Ion rifles stun and do massive damage via emps, and security even gets carbines to toss into a backpack. 

TL:DR
Secborgs can be tweaked in numerous ways, and for all the power given to them they have counters that damn near secure a Borg kill in a single click.

**Comment by Readystorm546** [comment]
And something i just member is RIP any donors who like the borg skins because they will just be null and void now. Also rip the people who play borg and wanna be sec. Now they cant do that. How sad

**Comment by nesisknesis** [comment]
> Most player are not fucking masters at the game and just because you or anyone you know is a master does not mean you should remove something that might help the newer players.

I realise this, however you do not need to be a master to play AI effectively, if you consider what you have, you'll find that the other modules meet the same standard, or more, as a security borg when emagged. For instance, a standard module can flash someone and decapitate them swiftly with their e-sword. And, for more insurance, an engie borg with their stunprod can tag along. The AI then just needs to wait for these resources to achieve this. By not requiring this tag-team, Sec borgs certainly make malf AI easier by excelling in tagging and bagging people, perhaps even too much, highlighting how powerful they are.  

The AI is an advanced role, malf AI more so, newer players will inevitably struggle but removing sec borgs won't make their lives impossible.

**Comment by nesisknesis** [comment]
> My ideas to tweak secborg are pretty simple. Let's look at why borgs are so powerful: they got their ranged disabler, and melee stunbaton. They also feel no pain, and fuckin zoom fast. Now, the only thing that makes them different from any normal security guard here is their speed and the fact they can go into negative health. Potential tweaks could include making the Borg slower then other non-combat borgs, or implement Borg slowdown when damaged type of system. And, of course, there's the fact that AI is seen as a way to validhunt by the players. Secborgs are, again, also the bread and butter of a malf ai. You could siphon, but pressure damage takes forever to kill and it feels like everyone carries a welder to weld down vents with now. Shock a door? Only works if they aren't carrying insuls, or a diagnostic hud, or walk into a door. And again, for all the powers a Borg has, its counters are deadly. For instance, the Borg cannot repair burn damage to other borgs, so lasers and fire will put borgs out or action for awhile. Flashes can be found and made in numerous locations, and instantly stun and mute a Borg for a considerable amount of time, and they include an aoe blind attack for you to get close enough for the stun. Ion rifles stun and do massive damage via emps, and security even gets carbines to toss into a backpack.
> 
> TL:DR
> Secborgs can be tweaked in numerous ways, and for all the power given to them they have counters that damn near secure a Borg kill in a single click.

These tweaks I support, borgs are undoubtedly powerful when played by a capable person. However, someone to code these changes can be a rarity which makes removal pretty much the most feasible solution, not an ideal one, but a solution. 

> Secborgs are, again, also the bread and butter of a malf ai.

An overstatement, you are underestimating the capability of the other modules which I touched on in my other comment. As a malf AI, I would much prefer an engie, standard or medical borg as my assassin, two have the capability to debraining, and one of disposing to space. Security borgs actually lack in doing these effectively. In a combat situation, security borgs would most likely fare better due to their ranged stun and laser capability. However, an AI's need for this is rare if they play their cards right. Keep in mind, other modules are combat efficient even while limited to melee ranged. You may argue that by removing the security module defeats silicon capability in combat as I argued before, but this should not be the route for AI's, and if it occurs they should take the consequence. Furthermore, plasma flooding is a solid weapon for station domination that substitutes for the lack of secborgs.

**Comment by nesisknesis** [comment]
> And something i just member is RIP any donors who like the borg skins because they will just be null and void now. Also rip the people who play borg and wanna be sec. Now they cant do that. How sad

I am willing to ask donors what new donors skins they would want as compensation.

**Comment by nesisknesis** [comment]
> This is gonna go just like xenoling, where we delete a big part of something without tweaking it or something first and it practically kills the thing. At least have something to replace it, or fuckin attempt to look at tweaks to it first. And what exactly is wrong with having combat oriented borgs? Perhaps it's because AIs frequently spawn with lawsets enabling them to valid, that this is an issue. Combat modules are also the bread and butter of antag AIs in many cases. And, perhaps, we can look at what makes the secborg so proficient at combat, and simply tweak that. You know how engiborgs have magnetic treads or whatever? Secborgs could just be slow. Makes it hard to valid the traitors if they can just outrun you. And is it the stunbaton that's the cause of the secborgs proficient combat skills? Why don't we look at that, instead of looking at something and deleting it instantly. I feel like we're treating a symptom and not the diease here.

My other comments relate to your arguments here. But what I find laughable is: 

EDIT: The removal of xenolings also did not kill xenobiology, the """rework""" did.

> where we delete a big part of something without tweaking it or something first and it practically kills the thing.

The security module is not a big part of silicons, and its deletion will not kill silicons. It is one of eight commonly available modules with probably the most specific intention for existence; to enforce space law by arresting wrongdoers. Other modules in comparison have varying tools that enable a wide range of activities to be performed, in contrast to the security module which has tools that enable it only to detain or maim. By nature of it being security, it is sadly used for validhunting, and if those who only play borg for the security module are saddened, so be it, the outcome will be that if they want to be security, they will need to play as a humanoid which has the same limitations that you so argue should be implemented for cyborgs. 

In my eyes, removal meets our quarrels with security borgs in a form of compromise. I see it as a medium of validhunting which will so be eliminated, and you see it as somewhat overpowered in the case that it does not have common drawbacks that humanoids do. By removal, players wanting to play security will inherently be forced to be a humanoid which have these drawbacks. I am not disregarding other modules still do not have these drawbacks, but it is removed in possibly the most prevalent module that benefits the most from it.

**Comment by Szyszkrzyneczka** [comment]
just fukken delete the fuckers

**Comment by monster860** [comment]
The problem with sec cyborgs is that they are a validhunt-only. At least with, say, a sec officer, the sec officer can do other things besides validhunting, like getting really drunk in the bar or redecorating security or something.  However, the secborg will be guaranteed to be extremely bored 30 minutes into the round, and will arrest anyone that even looks at them the wrong way.

**Comment by boba247** [comment]
This also hurts non rouge normal AI’s. For Asimov the AI needs a reliable intervention tool for preventing human harm, this falls to a sec Borg. Yes there are alternatives but their all shit. The AI could tell sec which isn’t bound by any laws to prevent human harm or listen to the AI so unless a murder is occurring there’s little chance sec will respond. A peacekeeper borg could go keep the peace, but this borg is basically an observer with tools that can barley deal with one threat. The standard borg might be the best bet with means to actually restrain someone and a flash yet beepsky could outperform this borg in effectiveness. Also generally to prevent human harm sending a very human sec officer into danger isn’t ideal, instead of sending a nonhuman cyborg with combat capabilities. 

As for a solution replace sec borgs with a larger version of beepsky, that’d look cool and even with just a stun baton and cuffs it’s more effective than any other borgs at combat. 

Or keep sec bogs as is and slow them down to less than running of a humanoid, but give them another item that boosts their speed for a ridiculous amount of power.  

Or maybe change our rules to make it clear valid hunting is a no no, and hand out punishments to reflect that. Technically it’s under power gaming right now. I assume valid hunting is harder to catch which is probably the argument against this, but I have never been an admin so wouldn’t know.

**Comment by Hopekz** [comment]
WIP "replacement" for when this PR goes through --> #7529

**Comment by Ratvar** [comment]
AI can use other humans to prevent human harm. Really, ask them, I heard they can do all stuff borgs can do. Summon beepsky. Bolt and depower doors. Use mediborg to save people. Actually use peaceborg who is actually good. Secborg removed all imagination from AI players.

**Comment by ynot01** [comment]
everyone will just move on to engieborg, the second best borg with a massive kit

what will you do about that?

**Comment by Ratvar** [comment]
Even then most of shitlers play secborg as shitcurity by default, not because they're maximizing validhunting power - otherwise they they would play engiborg from the start, for engiborg is plain better then secborgs due to selfrepair, welderbomb and RCD.
And then you can nerf engiborg's combat power.

**Comment by Readystorm546** [comment]
After seeing #7529 I am now "okay" with these changes but i still dont agree

**Comment by MegaEmpirical** [comment]
Don't delete the sec borgs. Disable them from being selected normally if we're replacing them, though I don't really have an issue with them.

**Comment by OrderoftheDarkOats** [comment]
This is the stupidest shit I've ever seen. "Imagine having a borg module that is so combat capable that all other borg modules are rendered useless" I mean no the other borgs aren't useless because their jobs are not combat. You shouldn't expect a med borg to saw people and for service borgs to hit with trays. They can do it but it isn't their purpose and they serve their intended function very well. If you're upset that every fucking unga who becomes a posi brain borg immediate becomes a sec borg then heres another option. If an AI is present they have to confirm their borgs module selection rather than let the borg lock it in themselves.

**Comment by SomeguyManperson** [comment]
you won't need other borgs if you can just validhunt the antag before they cause any damage

**Comment by Alexandria404** [comment]
"improve dont remove"
bad PR

**Comment by alexkar598** [comment]
img

Closing as per HC decision

**Comment by alexkar598** [comment]
For anyone interested: poll

**Comment by Ratvar** [comment]
Well of course people would fear removing shit features without any reason as to why, well of course server votes do not encourage people to discuss things, of course vote could be about replacing secborg with detective borg, and all other reason why making server vote at all was in bad faith
